File: 0119.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* getRow(int rowIndex, int* returnSize) {
    int table[34][34] = { 0 };
    int* ret;
    
    table[0][0] = 1;
    table[1][0] = 1;
    table[1][1] = 1;
    
    for (int i = 2; i <= rowIndex; i++) {
        for (int j = 0; j < i + 1; j++) {
            if (j == 0 || j == i) {
                table[i][j] = 1;
            }
            else {
                table[i][j] = table[i - 1][j - 1] + table[i - 1][j];
            }
        }
    }
        
    *returnSize = rowIndex + 1;
    ret = malloc((rowIndex + 1) * sizeof(int));
    memcpy(ret, &table[rowIndex], (rowIndex + 1) * sizeof(int));
    
    return ret;
}

File: 0119.py
class Solution(object):
    def getRow(self, rowIndex):
        """
        :type rowIndex: int
        :rtype: List[int]
        """
        ans = []
        
        # row 0
        row = []
        row.append(1)
        ans.append(row)
        if rowIndex == 0:
            return row
        
        # row 1
        row = []
        row.append(1)
        row.append(1)
        ans.append(row)
        
        if rowIndex == 1:
            return row
            
        i = 2
        while i <= rowIndex:
            j = 1
            row = [1]
            while j < i:
                row.append(ans[i - 1][j - 1] + ans[i - 1][j])
                j += 1
            row.append(1)
            ans.append(row)
            i += 1

        return row